Lib/test/test_posix.py

```diff
@@ -1166,7 +1166,7 @@ def test_mkdir_dir_fd(self):
             posix.close(f)
             support.rmtree(support.TESTFN + 'dir')
 
-    @unittest.skipUnless((os.mknod in os.supports_dir_fd) and hasattr(stat, 'S_IFIFO'),
+    @unittest.skipUnless(hasattr(os, 'mknod') and (os.mknod in os.supports_dir_fd) and hasattr(stat, 'S_IFIFO'),
```

Comment:
hmm I am a little confused here. According to documentation `os.supports_dif_fd`
says if a function is supported in the current platform [0]. 

If I'm not wrong when you run `hasattr(os, 'mknod')` always will be True, because
`os`, has the `mknod` method.

Please, could you say me where I'm wrong?

[0] https://docs.python.org/3.8/library/os.html?#os.supports_dir_fd

Reply:
os.supports_dir_fd lists functions that accept an open file descriptor for their dir_fd parameter. Something like below.

{<built-in function chown>, <built-in function mkdir>, <built-in function mknod>, <built-in function rmdir>, <built-in function mkfifo>, <built-in function rename>, <built-in function utime>, <built-in function link>, <built-in function open>, <built-in function readlink>, <built-in function symlink>, <built-in function unlink>, <built-in function chmod>, <built-in function stat>, <built-in function access>}

If some method, let's say os.mknod, is not supported at all by certain operating system, it will not be listed in os.supports_dir_fd.

Before we access the attribute mknod of os module, we have to ensure the os module has this attribute. Otherwise, the interpreter will report "module 'os' has no attribute 'mknod'" like below.

Traceback (most recent call last):
  File "/usr/lib/python3.8/runpy.py", line 192, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.8/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/usr/lib/python3.8/unittest/__main__.py", line 18, in <module>
    main(module=None)
  File "/usr/lib/python3.8/unittest/main.py", line 100, in __init__
    self.parseArgs(argv)
  File "/usr/lib/python3.8/unittest/main.py", line 147, in parseArgs
    self.createTests()
  File "/usr/lib/python3.8/unittest/main.py", line 158, in createTests
    self.test = self.testLoader.loadTestsFromNames(self.testNames,
  File "/usr/lib/python3.8/unittest/loader.py", line 220, in loadTestsFromNames
    suites = [self.loadTestsFromName(name, module) for name in names]
  File "/usr/lib/python3.8/unittest/loader.py", line 220, in <listcomp>
    suites = [self.loadTestsFromName(name, module) for name in names]
  File "/usr/lib/python3.8/unittest/loader.py", line 154, in loadTestsFromName
    module = __import__(module_name)
  File "/usr/lib/python3.8/test/test_posix.py", line 46, in <module>
    class PosixTester(unittest.TestCase):
  File "/usr/lib/python3.8/test/test_posix.py", line 1180, in PosixTester
    @unittest.skipUnless((os.mknod in os.supports_dir_fd) and hasattr(stat, 'S_IFIFO'),
AttributeError: module 'os' has no attribute 'mknod'